src/Controller/AbstractApiResourceController.php

```diff
@@ -140,14 +153,82 @@ public function getApiResourcesAction()
             in_array($orderByField, $this->getRepository()::getSortableFields())
             && in_array(strtolower($orderByDirection), ['asc', 'desc'])
         ) {
-            $queryBuilder->orderBy($queryBuilder->getRootAliases()[0] . '.' . $orderByField, $orderByDirection);
+            $queryBuilder->orderBy($queryBuilder->getRootAliases()[0].'.'.$orderByField, $orderByDirection);
         }
 
         $this->filterBuilderUpdater->addFilterConditions($form, $queryBuilder);
 
         return View::create($this->paginate($queryBuilder));
     }
 
+    /**
+     * @Rest\Get("/new")
+     * @param ParamFetcher $paramFetcher
+     * @param Request      $request
+     *
+     * @return Response
+     * @throws Exception
+     */
+    public function getNewApiResourceForm(ParamFetcher $paramFetcher, Request $request)
+    {
+        $router = $this->get("router");
```

Comment:
We can inject the RouterInterface in the class constructor. So we can also remove the AbstractFOSRestController inheritance.